### PR TITLE
Improved directive handling for places that use them

### PR DESCRIPTION
**Pull request by bbakerman**
#518 Improved directive handling for places that use them

### CODE REVIEW

File: src/main/java/graphql/language/Directive.java [hunk 1]
```diff
@@ -25,14 +30,38 @@ public Directive(String name, List<Argument> arguments) {
...
+        return getArgumentsMap().get(argumentName);
+    }
+
     public String getName() {
         return name;
     }
 
-    public void setName(String name) {
-        this.name = name;
+    /*
```
Comment (andimarek) [inline src/main/java/graphql/language/Directive.java]: Can we separate this helper from the actual class? For examples in a class `Directives` following JDK Conventions.

File: src/main/java/graphql/language/TypeDefinition.java [hunk 2]
```diff
@@ -14,4 +15,18 @@
      */
     List<Directive> getDirectives();
 
+    /**
```
Comment (andimarek) [inline src/main/java/graphql/language/TypeDefinition.java]: I am not sure about this: These methods do not add any real new value to the interface: they are just convenient.

If we really need/want an Interface with these methods we should add something like `DirectivesContainer` and make all classes implement is directly instead via `TypeDefinition`.

**Comment by bbakerman** [comment]
Ok I renamed the method, removed the TypeDefinition part.

I didnt create a Directives (plural) since there already was one.  I know packages could work.  Instead I created a NodesUtil helper

File: src/main/java/graphql/language/TypeDefinition.java (hunk 2 above)
Comment (bbakerman) [inline src/main/java/graphql/language/TypeDefinition.java]: removed

File: src/main/java/graphql/language/Directive.java (hunk 1 above)
Comment (bbakerman) [inline src/main/java/graphql/language/Directive.java]: changed

File: src/main/java/graphql/language/NodeUtil.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * Helper class for working with {@link Node}s
+ */
+@Internal
+public class NodeUtil {
+
+    public static Map<String, Directive> directivesByName(List<Directive> directives) {
+        return getByName(directives, Directive::getName);
+    }
+
+    public static Map<String, Argument> argumentByName(List<Argument> arguments) {
```
Comment (andimarek) [inline src/main/java/graphql/language/NodeUtil.java]: just a tiny things: `argumentsByName` (plural) :-)

**Pull request merged**